File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.java
class Solution {

    public int searchMinPos(int[] nums, int l, int r, int target){

        int mid = (l+r)/2;
        if(nums[mid] == target){
            return mid;
        }

        if(l>=r){
            return -1;
        }

        int num = searchMinPos(nums, l, mid, target);
        if(num == -1){
            return searchMinPos(nums, mid+1, r, target);
        }

        return num;

    }

    public int search(int[] nums, int target) {
        int l = 0;
        int min;
        int r = nums.length-1;

        int mid = (l+r)/2;
        
        int num = searchMinPos(nums, l, mid, target);
        if(num == -1){
            return searchMinPos(nums, mid+1, r,target);
        }

        return num;

    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        """
         how to get element at n-1 * r +c = r,c
         0,19 - c = 5, r 4
          10 = n-1*r +c =


          10 % t_c = c
          10 // t_c = r
        
        """
        r = len(matrix)

        c = len(matrix[0])

        i = 0 
        j = r*c -1
        

        while(i<=j):
            mid = (i+j)//2

            rx = mid//c
            cx = mid%c
            if matrix[rx][cx] == target:
                return True
            elif matrix[rx][cx] < target:
                i = mid+1
            else:
                j = mid-1
        return False
            
            


        
        